packages/api/src/graphql/query.private.ts

```diff
@@ -1141,6 +1142,29 @@ export const GraphQLQuery = new GraphQLObjectType<undefined, Context>({
           limit: Limit(first, last)
         })
       }
+    },
+    // Setting
+    // ======
+
+    setting: {
+      type: GraphQLSetting,
+      args: {id: {type: GraphQLID}, name: {type: GraphQLString}},
+      resolve(root, {id, name}, {authenticate, loaders}) {
+        // TODO authenticate
+
+        if ((id == null && name == null) || (id != null && name != null)) {
```

Comment:
We might think about using some kind of validations tool like Joi (https://joi.dev/api/?v=17.6.0) - something to think about for the future